src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -149,6 +156,16 @@ We support
             }
             else
             {
+                string rawRequirementType = configurationSection[RequirementTypeKeyword];
+
+                if (!string.IsNullOrEmpty(rawRequirementType))
+                {
+                    if (!Enum.TryParse(rawRequirementType, out requirementType))
+                    {
+                        throw new FormatException($"Invalid requirement type '{rawRequirementType}' for feature '{configurationSection.Key}'.");
```

Comment:
Do we use FormatException elsewhere? I'd like if we could use `FeatureManagementException`.

Reply:
Good call, I shouldn't introduce a new exception type. I switched it to `ArgumentException` instead.

Looking through our exceptions, all we have is-
```
FeatureManagementException
    FeatureManagementError.MissingFeatureFilter
    FeatureManagementError.AmbiguousFeatureFilter
    FeatureManagementError.MissingFeature
ArgumentNullException
ArgumentException
```

So I think ArgumentException makes the most sense. It's currently used by ValidateSettings in TargetingFilter.

A lot more error types get added by the v3 changes. Perhaps once those are back in, we can use the InvalidConfiguration error type. Or I can add it now if we want.